# Analyzing CIA Factbook Data Using SQL

In this project, we'll work with data from the CIA World Factbook, a compendium of statistics about all of the countries on Earth. 

The Factbook contains demographic information like:

- `population` - The population as of 2015.
- `population_growth` - The annual population growth rate, as a percentage.
- `area` - The total land and water area.

In [32]:
import sqlalchemy

sqlalchemy.create_engine('sqlite:///factbook.db')
%load_ext sql
%sql sqlite:///factbook.db

The sql extension is already loaded. To reload it, use:
  %reload_ext sql


'Connected: @factbook.db'

In [33]:
%%sql
SELECT *
  FROM sqlite_master
 WHERE type='table';

 * sqlite:///factbook.db
Done.


[('table', 'sqlite_sequence', 'sqlite_sequence', 3, 'CREATE TABLE sqlite_sequence(name,seq)'),
 ('table', 'facts', 'facts', 47, 'CREATE TABLE "facts" ("id" INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL, "code" varchar(255) NOT NULL, "name" varchar(255) NOT NULL, "area" integer, "a ... (4 characters truncated) ... land" integer, "area_water" integer, "population" integer, "population_growth" float, "birth_rate" float, "death_rate" float, "migration_rate" float)')]

## Data overview

Let's display the first 5 rows of the table `facts` and get familiar with the data.

In [34]:
%%sql
SELECT * 
  FROM facts 
 LIMIT 5;

 * sqlite:///factbook.db
Done.


[(1, 'af', 'Afghanistan', 652230, 652230, 0, 32564342, 2.32, 38.57, 13.89, 1.51),
 (2, 'al', 'Albania', 28748, 27398, 1350, 3029278, 0.3, 12.92, 6.58, 3.3),
 (3, 'ag', 'Algeria', 2381741, 2381741, 0, 39542166, 1.84, 23.67, 4.31, 0.92),
 (4, 'an', 'Andorra', 468, 468, 0, 85580, 0.12, 8.13, 6.96, 0.0),
 (5, 'ao', 'Angola', 1246700, 1246700, 0, 19625353, 2.78, 38.78, 11.49, 0.46)]

Here are the descriptions for some of the columns:

- `name` - The name of the country.
- `area` - The country's total area (both land and water).
- `area_land` - The country's land area in square kilometers.
- `area_water` - The country's waterarea in square kilometers.
- `population` - The country's population.
- `population_growth` - The country's population growth as a percentage.
- `birth_rate` - The country's birth rate, or the number of births a year per 1,000 people.
- `death_rate` - The country's death rate, or the number of death a year per 1,000 people.

Let's start by calculating some summary statistics and look for any outlier countries.

## Exploring Outliers

First, we will take a look at the maximum and minimum values for `population` and `population_growth`. 

In [35]:
%%sql 
SELECT MIN(population), MAX(population), 
       MIN(population_growth), MAX(population_growth)
  FROM facts;


 * sqlite:///factbook.db
Done.


[(0, 7256490011, 0.0, 4.02)]

A few things stick out from the summary statistics in the last screen:

- There's a country with a population of 0
- There's a country with a population of 7256490011 (or more than 7.2 billion people)

Let's use subqueries to zoom in on just these countries without using the specific values.

In [36]:
%%sql 
SELECT name, MAX(population)
  FROM facts;

 * sqlite:///factbook.db
Done.


[('World', 7256490011)]

In [37]:
%%sql 
SELECT name, MIN(population)
  FROM facts;

 * sqlite:///factbook.db
Done.


[('Antarctica', 0)]

It seems like the table contains a row for the whole world, which explains the population of over 7.2 billion. It also seems like the table contains a row for Antarctica, which explains the population of 0.

Now that we know this, we should recalculate the summary statistics we calculated earlier, while excluding the row for the whole world.

## Exploring Average Population and Area

In  this step we will calculate the summary statistics excluding `World`.

In [38]:
%%sql 
SELECT MIN(population), MAX(population), 
       MIN(population_growth), MAX(population_growth)
  FROM facts
 WHERE name != 'World';

 * sqlite:///factbook.db
Done.


[(0, 1367485388, 0.0, 4.02)]

In [39]:
%%sql 
SELECT AVG(population), AVG(area)
  FROM facts
 WHERE name != 'World';

 * sqlite:///factbook.db
Done.


[(32242666.56846473, 555093.546184739)]

## Finding Densely Populated Countries

To finish, we'll build on the query we wrote for the previous step to find countries that are densely populated. We'll identify countries that have:

 - Above average values for population.
 - Below average values for area.

In [40]:
%%sql 
SELECT name, population, area
  FROM facts
 WHERE name != 'World'
   AND population > (SELECT AVG(population)
                       FROM facts)
   AND area < (SELECT AVG(area)
                 FROM facts);

 * sqlite:///factbook.db
Done.


[('Bangladesh', 168957745, 148460),
 ('Germany', 80854408, 357022),
 ('Japan', 126919659, 377915),
 ('Philippines', 100998376, 300000),
 ('Thailand', 67976405, 513120),
 ('United Kingdom', 64088222, 243610),
 ('Vietnam', 94348835, 331210)]

## Finding most populated country 

In this step we will answer the following questions: 

- What country has the most people? 
- What country has the highest growth rate?


In [41]:
%%sql 
SELECT name, MAX(population)
  FROM facts
 WHERE name != 'World';

 * sqlite:///factbook.db
Done.


[('China', 1367485388)]

As expected, China is the most populated country in the world.

In [42]:
%%sql 
SELECT name, MAX(population_growth)
 FROM facts
 WHERE name != 'World';

 * sqlite:///factbook.db
Done.


[('South Sudan', 4.02)]

South Sudan has the highest growth rate. 

## Exploring water to land ratios 

We will find out:

- which countries have the highest ratios of water to land.
- which countries have more water than land. 

In order to  compute the ratios, we will divide the `area_water` by `area_land`.

In [43]:
%%sql 
SELECT name, area_water, area_land, 
       area_water/area_land AS water_land_ratio
  FROM facts
ORDER BY water_land_ratio DESC;

 * sqlite:///factbook.db
Done.


[('British Indian Ocean Territory', 54340, 60, 905),
 ('Virgin Islands', 1564, 346, 4),
 ('Afghanistan', 0, 652230, 0),
 ('Albania', 1350, 27398, 0),
 ('Algeria', 0, 2381741, 0),
 ('Andorra', 0, 468, 0),
 ('Angola', 0, 1246700, 0),
 ('Antigua and Barbuda', 0, 442, 0),
 ('Argentina', 43710, 2736690, 0),
 ('Armenia', 1540, 28203, 0),
 ('Australia', 58920, 7682300, 0),
 ('Austria', 1426, 82445, 0),
 ('Azerbaijan', 3971, 82629, 0),
 ('Bahamas, The', 3870, 10010, 0),
 ('Bahrain', 0, 760, 0),
 ('Bangladesh', 18290, 130170, 0),
 ('Barbados', 0, 430, 0),
 ('Belarus', 4700, 202900, 0),
 ('Belgium', 250, 30278, 0),
 ('Belize', 160, 22806, 0),
 ('Benin', 2000, 110622, 0),
 ('Bhutan', 0, 38394, 0),
 ('Bolivia', 15280, 1083301, 0),
 ('Bosnia and Herzegovina', 10, 51187, 0),
 ('Botswana', 15000, 566730, 0),
 ('Brazil', 157630, 8358140, 0),
 ('Brunei', 500, 5265, 0),
 ('Bulgaria', 2390, 108489, 0),
 ('Burkina Faso', 400, 273800, 0),
 ('Burma', 23070, 653508, 0),
 ('Burundi', 2150, 25680, 0),
 ('Cambodia', 4520, 176515, 0),
 ('Cameroon', 2730, 472710, 0),
 ('Canada', 891163, 9093507, 0),
 ('Cabo Verde', 0, 4033, 0),
 ('Central African Republic', 0, 622984, 0),
 ('Chad', 24800, 1259200, 0),
 ('Chile', 12290, 743812, 0),
 ('China', 270550, 9326410, 0),
 ('Colombia', 100210, 1038700, 0),
 ('Comoros', 0, 2235, 0),
 ('Congo, Democratic Republic of the', 77810, 2267048, 0),
 ('Congo, Republic of the', 500, 341500, 0),
 ('Costa Rica', 40, 51060, 0),
 ("Cote d'Ivoire", 4460, 318003, 0),
 ('Croatia', 620, 55974, 0),
 ('Cuba', 1040, 109820, 0),
 ('Cyprus', 10, 9241, 0),
 ('Czech Republic', 1620, 77247, 0),
 ('Denmark', 660, 42434, 0),
 ('Djibouti', 20, 23180, 0),
 ('Dominica', 0, 751, 0),
 ('Dominican Republic', 350, 48320, 0),
 ('Ecuador', 6720, 276841, 0),
 ('Egypt', 6000, 995450, 0),
 ('El Salvador', 320, 20721, 0),
 ('Equatorial Guinea', 0, 28051, 0),
 ('Eritrea', 16600, 101000, 0),
 ('Estonia', 2840, 42388, 0),
 ('Fiji', 0, 18274, 0),
 ('Finland', 34330, 303815, 0),
 ('France', 3374, 640427, 0),
 ('Gabon', 10000, 257667, 0),
 ('Gambia, The', 1180, 10120, 0),
 ('Georgia', 0, 69700, 0),
 ('Germany', 8350, 348672, 0),
 ('Ghana', 11000, 227533, 0),
 ('Greece', 1310, 130647, 0),
 ('Grenada', 0, 344, 0),
 ('Guatemala', 1730, 107159, 0),
 ('Guinea', 140, 245717, 0),
 ('Guinea-Bissau', 8005, 28120, 0),
 ('Guyana', 18120, 196849, 0),
 ('Haiti', 190, 27560, 0),
 ('Honduras', 200, 111890, 0),
 ('Hungary', 3420, 89608, 0),
 ('Iceland', 2750, 100250, 0),
 ('India', 314070, 2973193, 0),
 ('Indonesia', 93000, 1811569, 0),
 ('Iran', 116600, 1531595, 0),
 ('Iraq', 950, 437367, 0),
 ('Ireland', 1390, 68883, 0),
 ('Israel', 440, 20330, 0),
 ('Italy', 7200, 294140, 0),
 ('Jamaica', 160, 10831, 0),
 ('Japan', 13430, 364485, 0),
 ('Jordan', 540, 88802, 0),
 ('Kazakhstan', 25200, 2699700, 0),
 ('Kenya', 11227, 569140, 0),
 ('Kiribati', 0, 811, 0),
 ('Korea, North', 130, 120408, 0),
 ('Korea, South', 2800, 96920, 0),
 ('Kosovo', 0, 10887, 0),
 ('Kuwait', 0, 17818, 0),
 ('Kyrgyzstan', 8150, 191801, 0),
 ('Laos', 6000, 230800, 0),
 ('Latvia', 2340, 62249, 0),
 ('Lebanon', 170, 10230, 0),
 ('Lesotho', 0, 30355, 0),
 ('Liberia', 15049, 96320, 0),
 ('Libya', 0, 1759540, 0),
 ('Liechtenstein', 0, 160, 0),
 ('Lithuania', 2620, 62680, 0),
 ('Luxembourg', 0, 2586, 0),
 ('Macedonia', 280, 25433, 0),
 ('Madagascar', 5501, 581540, 0),
 ('Malawi', 24404, 94080, 0),
 ('Malaysia', 1190, 328657, 0),
 ('Maldives', 0, 298, 0),
 ('Mali', 20002, 1220190, 0),
 ('Malta', 0, 316, 0),
 ('Marshall Islands', 0, 181, 0),
 ('Mauritania', 0, 1030700, 0),
 ('Mauritius', 10, 2030, 0),
 ('Mexico', 20430, 1943945, 0),
 ('Micronesia, Federated States of', 0, 702, 0),
 ('Moldova', 960, 32891, 0),
 ('Monaco', 0, 2, 0),
 ('Mongolia', 10560, 1553556, 0),
 ('Montenegro', 360, 13452, 0),
 ('Morocco', 250, 446300, 0),
 ('Mozambique', 13000, 786380, 0),
 ('Namibia', 1002, 823290, 0),
 ('Nauru', 0, 21, 0),
 ('Nepal', 3830, 143351, 0),
 ('Netherlands', 7650, 33893, 0),
 ('Nicaragua', 10380, 119990, 0),
 ('Niger', 

In [44]:
%%sql 
SELECT name, area_water, area_land
  FROM facts 
 WHERE area_water > area_land;

 * sqlite:///factbook.db
Done.


[('British Indian Ocean Territory', 54340, 60), ('Virgin Islands', 1564, 346)]

As we can see, British Indian Ocean Territory has more water than land along with Virgin Islands as well as the highest water to land ratio. 

## Exploring birth and migration rates

To answer the question which countries will add the most people to their population next year, we need to find countries with the highest value of both birth rate and migration rate. 

In [45]:
%%sql 
SELECT name, ROUND(birth_rate + migration_rate) AS most_people_added
  FROM facts
ORDER BY most_people_added DESC;

 * sqlite:///factbook.db
Done.


[('Somalia', 49.0),
 ('South Sudan', 48.0),
 ('Mali', 47.0),
 ('Niger', 46.0),
 ('Uganda', 45.0),
 ('American Samoa', 44.0),
 ('Sao Tome and Principe', 43.0),
 ('Zambia', 43.0),
 ('Burkina Faso', 42.0),
 ('Burundi', 42.0),
 ('Congo, Republic of the', 42.0),
 ('Malawi', 42.0),
 ('Syria', 42.0),
 ('Micronesia, Federated States of', 41.0),
 ('Mozambique', 41.0),
 ('Tonga', 41.0),
 ('Afghanistan', 40.0),
 ('Chad', 40.0),
 ('Sierra Leone', 40.0),
 ('Angola', 39.0),
 ('Jordan', 39.0),
 ('Nauru', 39.0),
 ('Nigeria', 38.0),
 ('Timor-Leste', 38.0),
 ('Ethiopia', 37.0),
 ('Gabon', 37.0),
 ('Tanzania', 37.0),
 ('Benin', 36.0),
 ('Cameroon', 36.0),
 ('Guinea', 36.0),
 ('Senegal', 36.0),
 ('Central African Republic', 35.0),
 ('Congo, Democratic Republic of the', 35.0),
 ('Rwanda', 35.0),
 ('Liberia', 34.0),
 ('Togo', 34.0),
 ('Equatorial Guinea', 33.0),
 ('Gambia, The', 33.0),
 ('Ghana', 33.0),
 ('Guinea-Bissau', 33.0),
 ('Iraq', 33.0),
 ('Lesotho', 33.0),
 ('Madagascar', 33.0),
 ('Sudan', 33.0),
 ('Mauritania', 32.0),
 ('Qatar', 32.0),
 ('Zimbabwe', 32.0),
 ('Samoa', 31.0),
 ('Tuvalu', 31.0),
 ('Yemen', 31.0),
 ('Gaza Strip', 31.0),
 ('Comoros', 30.0),
 ('Djibouti', 30.0),
 ('Eritrea', 30.0),
 ('Marshall Islands', 30.0),
 ("Cote d'Ivoire", 29.0),
 ('Luxembourg', 29.0),
 ('Kyrgyzstan', 28.0),
 ('Maldives', 28.0),
 ('Solomon Islands', 28.0),
 ('United Arab Emirates', 28.0),
 ('British Virgin Islands', 28.0),
 ('Bahrain', 27.0),
 ('Guatemala', 27.0),
 ('Kenya', 27.0),
 ('Vanuatu', 27.0),
 ('Cayman Islands', 27.0),
 ('Botswana', 26.0),
 ('Fiji', 26.0),
 ('Libya', 26.0),
 ('Philippines', 26.0),
 ('Tajikistan', 26.0),
 ('Turks and Caicos Islands', 26.0),
 ('Algeria', 25.0),
 ('Belize', 25.0),
 ('El Salvador', 25.0),
 ('Haiti', 25.0),
 ('Laos', 25.0),
 ('Nepal', 25.0),
 ('Oman', 25.0),
 ('Swaziland', 25.0),
 ('Anguilla', 25.0),
 ('Northern Mariana Islands', 25.0),
 ('Cambodia', 24.0),
 ('Guyana', 24.0),
 ('Honduras', 24.0),
 ('Kiribati', 24.0),
 ('Pakistan', 24.0),
 ('Papua New Guinea', 24.0),
 ('Bolivia', 23.0),
 ('Egypt', 23.0),
 ('Jamaica', 23.0),
 ('Saint Vincent and the Grenadines', 23.0),
 ('South Africa', 23.0),
 ('Guam', 23.0),
 ('West Bank', 23.0),
 ('Bangladesh', 22.0),
 ('Moldova', 22.0),
 ('Morocco', 22.0),
 ('Singapore', 22.0),
 ('Cabo Verde', 21.0),
 ('Cyprus', 21.0),
 ('Dominica', 21.0),
 ('Dominican Republic', 21.0),
 ('Israel', 21.0),
 ('Kuwait', 21.0),
 ('Mongolia', 21.0),
 ('Nicaragua', 21.0),
 ('Peru', 21.0),
 ('Turkmenistan', 21.0),
 ('Aruba', 21.0),
 ('Brunei', 20.0),
 ('India', 20.0),
 ('Kazakhstan', 20.0),
 ('Malaysia', 20.0),
 ('Mexico', 20.0),
 ('Namibia', 20.0),
 ('Trinidad and Tobago', 20.0),
 ('Greenland', 20.0),
 ('Sint Maarten', 20.0),
 ('Armenia', 19.0),
 ('Burma', 19.0),
 ('Ecuador', 19.0),
 ('Grenada', 19.0),
 ('Ireland', 19.0),
 ('Norway', 19.0),
 ('Panama', 19.0),
 ('Saudi Arabia', 19.0),
 ('Uzbekistan', 19.0),
 ('Venezuela', 19.0),
 ('New Caledonia', 19.0),
 ('Wallis and Futuna', 19.0),
 ('Puerto Rico', 19.0),
 ('Antigua and Barbuda', 18.0),
 ('Australia', 18.0),
 ('Bhutan', 18.0),
 ('Iceland', 18.0),
 ('Indonesia', 18.0),
 ('Iran', 18.0),
 ('Spain', 18.0),
 ('Tunisia', 18.0),
 ('Turkey', 18.0),
 ('Isle of Man', 18.0),
 ('Virgin Islands', 18.0),
 ('Argentina', 17.0),
 ('Azerbaijan', 17.0),
 ('Belgium', 17.0),
 ('Colombia', 17.0),
 ('Costa Rica', 17.0),
 ('Saint Lucia', 17.0),
 ('San Marino', 17.0),
 ('Sri Lanka', 17.0),
 ('Suriname', 17.0),
 ('Sweden', 17.0),
 ('Gibraltar', 17.0),
 ('Albania', 16.0),
 ('Bahamas, The', 16.0),
 ('Canada', 16.0),
 ('Latvia', 16.0),
 ('Lebanon', 16.0),
 ('Liechtenstein', 16.0),
 ('Lithuania', 16.0),
 ('New Zealand', 16.0),
 ('Paraguay', 16.0),
 ('United States', 16.0),
 ('Vietnam', 16.0),
 ('French Polynesia', 16.0),
 ('Saint Pierre and Miquelon', 16.0),
 ('Jersey', 16.0),
 ('Austria', 15.0),
 ('Brazil', 15.0),
 ('Georgia', 15.0),
 ('Korea, North', 15.0),
 ('Saint Kitts and Nevis', 15.0),
 ('Seychelles', 15.0),
 ('Switzerland', 15.0),
 ('United Kingdom', 15.0),
 ('Curacao', 15.0)

As we can see, Somalia, South Sudan and Mali are most likely to add more people to its population. 

## Death Rate VS Birth Rate

In this step we will identify which countries have a higher death rate than birth rate. 

In [46]:
%%sql 
SELECT name, birth_rate, death_rate
  FROM facts
 WHERE death_rate > birth_rate;

 * sqlite:///factbook.db
Done.


[('Austria', 9.41, 9.42),
 ('Belarus', 10.7, 13.36),
 ('Bosnia and Herzegovina', 8.87, 9.75),
 ('Bulgaria', 8.92, 14.44),
 ('Croatia', 9.45, 12.18),
 ('Czech Republic', 9.63, 10.34),
 ('Estonia', 10.51, 12.4),
 ('Germany', 8.47, 11.42),
 ('Greece', 8.66, 11.09),
 ('Hungary', 9.16, 12.73),
 ('Italy', 8.74, 10.19),
 ('Japan', 7.93, 9.51),
 ('Latvia', 10.0, 14.31),
 ('Lithuania', 10.1, 14.27),
 ('Moldova', 12.0, 12.59),
 ('Monaco', 6.65, 9.24),
 ('Poland', 9.74, 10.19),
 ('Portugal', 9.27, 11.02),
 ('Romania', 9.14, 11.9),
 ('Russia', 11.6, 13.69),
 ('Serbia', 9.08, 13.66),
 ('Slovenia', 8.42, 11.37),
 ('Ukraine', 10.72, 14.46),
 ('Saint Pierre and Miquelon', 7.42, 9.72)]

The table above represents countires where death rate is higher than birth rate. 

## Exploring Population/Area Ratio

What countries have the highest population/area ratio and how does it compare to list of densely populated counties from the previous step? Let's find out.

In [47]:
%%sql 
SELECT name, population/area AS population_area_ratio
  FROM facts 
ORDER BY population_area_ratio DESC;

 * sqlite:///factbook.db
Done.


[('Macau', 21168),
 ('Monaco', 15267),
 ('Singapore', 8141),
 ('Hong Kong', 6445),
 ('Gaza Strip', 5191),
 ('Gibraltar', 4876),
 ('Bahrain', 1771),
 ('Maldives', 1319),
 ('Malta', 1310),
 ('Bermuda', 1299),
 ('Sint Maarten', 1167),
 ('Bangladesh', 1138),
 ('Guernsey', 847),
 ('Jersey', 838),
 ('Barbados', 675),
 ('Mauritius', 656),
 ('Taiwan', 650),
 ('Aruba', 623),
 ('Lebanon', 594),
 ('Saint Martin', 588),
 ('San Marino', 541),
 ('Korea, South', 492),
 ('Rwanda', 480),
 ('West Bank', 475),
 ('Nauru', 454),
 ('Tuvalu', 418),
 ('Netherlands', 407),
 ('Marshall Islands', 398),
 ('Israel', 387),
 ('Burundi', 385),
 ('India', 380),
 ('Belgium', 370),
 ('Haiti', 364),
 ('Comoros', 349),
 ('Philippines', 336),
 ('Sri Lanka', 336),
 ('Japan', 335),
 ('Curacao', 334),
 ('Grenada', 321),
 ('Guam', 297),
 ('El Salvador', 291),
 ('Vietnam', 284),
 ('American Samoa', 273),
 ('Jamaica', 268),
 ('Saint Lucia', 266),
 ('Saint Vincent and the Grenadines', 263),
 ('United Kingdom', 263),
 ('Puerto Rico', 260),
 ('Pakistan', 250),
 ('Trinidad and Tobago', 238),
 ('Liechtenstein', 235),
 ('Germany', 226),
 ('British Virgin Islands', 221),
 ('Luxembourg', 220),
 ('Dominican Republic', 215),
 ('Nepal', 214),
 ('Cayman Islands', 212),
 ('Antigua and Barbuda', 209),
 ('Korea, North', 207),
 ('Italy', 205),
 ('Seychelles', 203),
 ('Sao Tome and Principe', 201),
 ('Saint Kitts and Nevis', 198),
 ('Nigeria', 196),
 ('Switzerland', 196),
 ('Qatar', 189),
 ('Andorra', 182),
 ('Anguilla', 180),
 ('Gambia, The', 174),
 ('Kosovo', 171),
 ('Kuwait', 156),
 ('Uganda', 153),
 ('Isle of Man', 153),
 ('Malawi', 151),
 ('Micronesia, Federated States of', 149),
 ('China', 142),
 ('Tonga', 142),
 ('Guatemala', 137),
 ('Cabo Verde', 135),
 ('Czech Republic', 134),
 ('Indonesia', 134),
 ('Thailand', 132),
 ('Togo', 132),
 ('Kiribati', 130),
 ('Denmark', 129),
 ('Cyprus', 128),
 ('Akrotiri', 127),
 ('Poland', 123),
 ('Dhekelia', 120),
 ('European Union', 118),
 ('Portugal', 117),
 ('Azerbaijan', 112),
 ('Northern Mariana Islands', 112),
 ('Slovakia', 111),
 ('Tokelau', 111),
 ('Ghana', 110),
 ('Wallis and Futuna', 109),
 ('Hungary', 106),
 ('Albania', 105),
 ('Moldova', 104),
 ('Austria', 103),
 ('France', 103),
 ('Armenia', 102),
 ('Turkey', 101),
 ('Cuba', 99),
 ('Dominica', 98),
 ('Slovenia', 97),
 ('Spain', 95),
 ('Costa Rica', 94),
 ('Benin', 92),
 ('Malaysia', 92),
 ('Serbia', 92),
 ('Syria', 92),
 ('Ethiopia', 90),
 ('Jordan', 90),
 ('Romania', 90),
 ('Egypt', 88),
 ('Cambodia', 86),
 ('Iraq', 84),
 ('Burma', 83),
 ('Swaziland', 82),
 ('Timor-Leste', 82),
 ('Greece', 81),
 ('Macedonia', 81),
 ('Sierra Leone', 81),
 ('Kenya', 79),
 ('Croatia', 78),
 ('Honduras', 78),
 ('Bosnia and Herzegovina', 75),
 ('Brunei', 74),
 ('Morocco', 74),
 ('Ukraine', 73),
 ("Cote d'Ivoire", 72),
 ('Senegal', 71),
 ('Georgia', 70),
 ('Burkina Faso', 69),
 ('Ireland', 69),
 ('Samoa', 69),
 ('United Arab Emirates', 69),
 ('Tunisia', 67),
 ('French Polynesia', 67),
 ('Uzbekistan', 65),
 ('Bulgaria', 64),
 ('Lesotho', 64),
 ('Mexico', 61),
 ('Norfolk Island', 61),
 ('Tajikistan', 56),
 ('Ecuador', 55),
 ('Eritrea', 55),
 ('Virgin Islands', 54),
 ('Tanzania', 53),
 ('Turks and Caicos Islands', 53),
 ('Montserrat', 51),
 ('Yemen', 50),
 ('Afghanistan', 49),
 ('Cameroon', 49),
 ('Fiji', 49),
 ('Iran', 49),
 ('Panama', 48),
 ('Guinea', 47),
 ('Guinea-Bissau', 47),
 ('Belarus', 46),
 ('Montenegro', 46),
 ('Palau', 46),
 ('Nicaragua', 45),
 ('Lithuania', 44),
 ('South Africa', 44),
 ('Cocos (Keeling) Islands', 42),
 ('Colombia', 41),
 ('Cook Islands', 41),
 ('Madagascar', 40),
 ('Liberia', 37),
 ('Zimbabwe', 36),
 ('Faroe Islands', 36),
 ('Djibouti', 35),
 ('Congo, Democratic Republic of the', 33),
 ('United States', 32),
 ('Venezuela', 32),
 ('Mozambique', 31),
 ('Latvia', 30),
 ('Laos', 29),
 ('Kyrgyzstan', 28),
 ('Estonia', 27),
 ('Equatorial Guinea', 26),
 ('Saint Helena, Ascension, and Tristan da Cunha', 25),
 ('Bahamas, The', 23),
 ('Brazil', 23),
 ('Chile', 23),
 ('Peru', 23),
 ('Saint


Countries with the highest population to area ratio are Macau, Monaco and Singapore. Whereas, countries with population higher than the average population, and the area smaller than the average area are Bangladesh, Germany, and Japan. The results  are different for obvious reasons, in the first case we compare the ratios of individual countries between each other, and in the second case, we compare each country's population and area values against the average of all countries. 

# Conclusion 

In this project, we analyzed the CIA Factbook Data containing the demographics of all countries in the world. We explored countries with highest/lowest population and population growth, found densely populated countries, calculated the water/land ratio and explored countries that have more water than land. We used SQL queries to accomplish all tasks including aggregate functions, 'group by' clause to compute summary statistics by unique value, nested subqueries and others. 